### ISSUE TITLE
`use_key_in_widget_constructors` should report only if super has a constructor with `key`

### ISSUE THREAD

**Issue by asashour** [dart-lang/linter#3384]
case 1:
```dart
import 'package:flutter/material.dart';

class ParentWidget extends StatelessWidget { // the lint here is ok
  @override
  Widget build(BuildContext context) => Container();
}

class MyWidget extends ParentWidget { // <--- the lint here is wrong
}
```

case 2:
```dart
import 'package:flutter/material.dart';

class ParentWidget extends StatelessWidget { // the lint here is ok
  int? i;

  ParentWidget({this.i});

  @override
  Widget build(BuildContext context) => Container();
}

class MyWidget extends ParentWidget { // <--- the lint here is wrong
}
```

**Expected behavior**
In `MyWidget`, there shouldn't be a lint, because the parent constructor doesn't have a `key`.

**Actual behavior**
In `MyWidget`, there is a lint reported.

**Comment by srawlins** [comment]
CC @goderbauer WDYT?

**Comment by goderbauer** [comment]
I am not so sure about this one. The lint is there to remind people that every public widget should have a key parameter. So, linting on `MyWidget` seems reasonable. It should force you to wire the key through to `ParentWidget` and fix that it doesn't expose a key. This is especially important if the parent class happens to be private:

```dart
class _Foo extends StatelessWidget {
  const _Foo(); // no lint, because its private

  @override
  Widget build(BuildContext context) {
    return Container();
  }
}

class Bar extends _Foo { // <--- should lint to force you to wire through the key
  
}
```

**Comment by bwilkerson** [comment]
I don't disagree with you, but we'd probably need to not lint in cases where the parent widget isn't under the user's control, as in those cases there isn't any way for the user to correctly fix the problem.

**Comment by goderbauer** [comment]
Ideally, you would take the lint as encouragement to file an issue against whoever owns the parent to wire through the key. :)

But yeah, I can see how that can become tiresome. I wouldn't be totally opposed to silencing the lint if fixing it is not in your control.
